Publish Magento Cloud 2FA for SSH beta documentation (#7517)
magento/devdocs: src/cloud/project/user-admin.md

| **Super user** | Project | Administrator access to all project settings and Cloud environments. Super users<br>can change settings and perform tasks on any environment, including creating<br>and restoring [snapshots][]. |
| **Project reader** | Project | View access to all project environments. Users with this role cannot perform tasks on<br>any environment. However, you can configure environment-level permissions for users with<br>this role to permit write access to a specific environment. |
| **Admin** | Environment | Change settings and perform tasks on an environment, including merging with the parent<br>environment |
| **Contributor** | Environment | Push code and branch the environment |
| **Reader** | Environment | View-only access to an environment |

## Add user authentication requirements

For added security, Magento is introducing a project-level MFA enforcement feature to improve security for SSH access to Cloud infrastructure. See [MFA enforcement for SSH].

> When MFA enforcement is enabled on a {{site.data.var.ece}} project, all users that with SSH access to an environment in that project must enable 2FA on their {{site.data.var.ece}} account. For automated processes, users must create an API token that machine users can use to authenticate from the command line. See [Enable user accounts for 2FA and SSH access](#update-account-security-settings).

Author: ```suggestion
When MFA enforcement is enabled on a {{site.data.var.ece}} project, all users with SSH access to an environment in that project must enable 2FA on their {{site.data.var.ece}} account. For automated processes, users must create an API token that machine users can use to authenticate. See [Enable user accounts for 2FA and SSH access](#update-account-security-settings).
```

{%include cloud/note-prevent-site-availability-issues.md%}

## Update account security settings

> After you add a user to a Cloud project, ask the user to review their account settings and add the following security configuration as needed:

Author: ```suggestion
After you add a user to a Cloud project, ask them to review their account settings and add the following security configuration as needed:
```

-  Enable two-factor authentication (TFA)

>    Magento recommends adding two-factor authentication to all accounts to meet security and compliance standards. Some projects will soon require two-factor authentication for all accounts that use SSH to connect to {{site.data.var.ece}} projects.

Author: ```suggestion
   Magento recommends adding two-factor authentication to all accounts to meet security and compliance standards. Some projects will soon require two-factor authentication on all accounts that use SSH to connect to {{site.data.var.ece}} projects.
```

-  Enable SSH keys

   Users that require access to {{site.data.var.ece}} source code repositories and infrastructure must enable SSH keys on their account. See [Enable SSH keys][].

-  Create an API token

>    You can generate an API token on your account that can be used for secure machine access to an environment through an SSH connection. You can use the API token to enable login workflows for automated processes.

Author: ```suggestion
   You can generate an API token on your account that can be used for secure machine access to an environment through an SSH connection. Use the API token to enable login workflows for automated processes.
```

   ![Cloud 2FA app device setup]({{ site.baseurl }}/common/images/cloud/cloud-2fa-settings-tfa-app-example.png){:width="400px"}

1. On the _TFA set up - Application_ page, type the two-factor authentication code from your mobile device in the **Application verification code** field.

   ![Cloud 2FA app device setup]({{ site.baseurl }}/common/images/cloud/cloud_account_settings-tfa-setup-steps.png){:width="550px"}

1. Click **Verify and save**.

1. Optional. Enable _Trusted browser_ settings to save current authentication code in the browser session for 30 days and reduce number of authentication challenges during project log in.

1. Click **Save** or **Skip**.

1. On the _TFA setup - Recovery_ codes page, copy and save the recovery codes so that you can log into your {{site.data.var.ece}} project when you cannot access your mobile device or authentication app.

   ![Cloud TFA recovery codes]({{ site.baseurl }}/common/images/cloud/cloud-2fa-settings-tfa-recovery-codes.png){:width="550px"}

   {:.bs.callout-warning}

>    Copy the recovery codes to another location or write them down to use if you lose access to your device or authentication app. Save the codes to your account to view and manage them from your account security settings. If you lose account access on a 2FA-enabled account and have no recovery codes, you must contact your Project administrator, or submit a Magento support ticket to reset the TFA application.

Author: ```suggestion
   Copy the recovery codes to another location or write them down to recover access if you your device or access to the authenticator application. Save the codes to your account to view and manage them from your account security settings. If you lose account access on a 2FA-enabled account and have no recovery codes, you must contact your Project administrator, or submit a Magento support ticket to reset the TFA application.
```

   On projects with the MFA enforcement feature enabled, you must use the API token to authenticate SSH access requests from automated accounts. to bypass authentication workflows which require 2FA.

### Enable 2FA

{{site.data.var.ece}} supports 2FA using any of the following applications:

-  [Google Authenticator (Android/iPhone/BlackBerry)][]
-  [Authy (Android/iPhone)][]
-  [Authenticator (Windows Phone)][]
-  [FreeOTP (Android)][]
-  [GAuth Authenticator (Firefox OS, desktop, others)][]

You can find instructions for setting up the authenticator app and enabling 2FA on the _Account settings_ page for your {{site.data.var.ece}} user account.

{:.procedure}
To enable 2FA on your {{site.data.var.ece}} user account:

1. Log in to the [{{site.data.var.ece}} user account][{{site.data.var.ece}} account].

1. On the Cloud projects page, click the **Account settings** tab.

   ![Cloud projects page]({{ site.baseurl }}/common/images/cloud/cloud-account-settings-tab.png){:width="550px"}

1. On the Account settings tab, expand the **Security** section. Then, click **Set up application**.

>    ![Cloud Security settings]({{ site.baseurl }}/common/images/cloud/
>    cloud_account_settings-tfa-setup-app.png){:width="550px"}

Reviewer: ```suggestion
   ![Cloud Security settings]({{ site.baseurl }}/common/images/cloud/cloud_account_settings-tfa-setup-app.png){:width="550px"}
```

1. Use the linked instructions to install one of the approved authentication applications on your mobile device, if needed.

>    ![Cloud Security settings]({{ site.baseurl }}/common/images/cloud/
>    cloud_account_settings-tfa-setup-app-options.png){:width="550px"}

Reviewer: ```suggestion
   ![Cloud Security settings]({{ site.baseurl }}/common/images/cloud/cloud_account_settings-tfa-setup-app-options.png){:width="550px"}
```

Author: Thanks for the suggestions-- I didn't see them before building and pushing from local. Build running now 🤞
